corehq/apps/ota/utils.py

```diff
@@ -87,7 +90,50 @@ def demo_restore_date_created(commcare_user):
             return restore.timestamp_created
 
 
-def is_permitted_to_restore(domain, couch_user, as_user, has_data_cleanup_privelege):
+def _ensure_valid_domain(domain, couch_user):
+    if not couch_user.is_member_of(domain):
+        raise RestorePermissionDenied(u"{} was not in the domain {}".format(couch_user.username, domain))
+
+
+def _ensure_cleanup_permission(domain, couch_user, as_user, has_data_cleanup_privilege):
+    if not has_data_cleanup_privilege and not couch_user.is_superuser:
+        raise RestorePermissionDenied(u"{} does not have permissions to restore as {}".format(
+            couch_user.username,
+            as_user,
+        ))
+
+
+def _ensure_valid_restore_as_user(domain, couch_user, as_user):
+    try:
+        username = as_user.split('@')[0]
+        user_domain = as_user.split('@')[1]
+    except IndexError:
+        raise RestorePermissionDenied(
+            u"Invalid restore user {}. Format is <user>@<domain>".format(as_user)
+        )
+    else:
+        if user_domain != domain:
+            # In this case we may be dealing with a WebUser
+            user = WebUser.get_by_username(as_user)
+            if not user or not user.is_member_of(domain):
+                raise RestorePermissionDenied(u"{} was not in the domain {}".format(username, domain))
+
+
+def _ensure_accessible_location(domain, couch_user, as_user):
+    if couch_user.is_commcare_user():
```

Comment:
i think so, but i didn't think locations was really supported for web users. @esoergel is that true? i thought locations were only configurable for mobile workers